app/processing/payment/get-schedule.js

```diff
@@ -0,0 +1,23 @@
+const getPeriodFromSchedule = require('./get-period-from-schedule')
+
+const getSchedule = (scheduleDate, totalPayments, paymentValue, settledValue, previousSettledValue, totalValue, increment, unit, settlementDate) => {
+  const scheduleDates = []
+  let expectedSettlementValue = 0
+  for (let i = 1; i <= totalPayments; i++) {
+    expectedSettlementValue = getExpectedValue(totalValue, totalPayments, i)
+    const cappedSettlementValue = settledValue <= expectedSettlementValue ? settledValue : expectedSettlementValue
```

Comment:
Need to cap the settlement value because BACS rejections will result in incorrect settlement values that may mean the schedule thinks more has been paid that actually has.

So if we cap the amount to be the maximum possible by the settlement date.